contracts/IdentityManager.sol

```diff
@@ -0,0 +1,98 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.20;
+
+import "@rarimo/evm-bridge-contracts/utils/Signers.sol";
+
+import "./interfaces/IIdentityManager.sol";
+
+contract IdentityManager is IIdentityManager, Signers {
+    uint256 public constant ROOT_EXPIRATION_TIME = 1 hours;
+
+    address public sourceStateContract;
+
+    uint256 internal _latestRoot;
+    uint256 internal _latestTimestamp;
+
+    mapping(uint256 => RootData) internal _roots;
+
+    function __IdentityManager_init(
+        address signer_,
+        address sourceStateContract_,
+        string calldata chainName_
+    ) external initializer {
+        __Signers_init(signer_, chainName_);
+
+        sourceStateContract = sourceStateContract_;
+    }
+
+    function signedTransitRoot(
+        uint256 prevRoot_,
+        uint256 postRoot_,
+        uint256 replacedAt_,
+        bytes calldata proof_
+    ) external override {
+        _checkMerkleSignature(_getSignHash(prevRoot_, postRoot_, replacedAt_), proof_);
```

Comment:
I would probably add a require here to check if `prevRoot_` has been already submitted. Like

`require(_roots[prevRoot_].replacedBy == bytes32(0), "IdentityManager: can't update already stored root")`